src/audio/dai.c

```diff
@@ -192,6 +243,11 @@ static void dai_free(struct comp_dev *dev)
 {
 	struct dai_data *dd = comp_get_drvdata(dev);
 
+	if (dd->group) {
+		notifier_unregister(dev, dd->group, NOTIFIER_ID_DAI_TRIGGER);
+		dd->group->num_dais--;
```

Comment:
ALso need to delete from the group list?

Reply:
The group list doesn't store any pointers to DAIs. Only number is stored.